maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java

```diff
@@ -1733,4 +1746,21 @@ private static Severity getSeverity(int validationLevel, int errorThreshold) {
             return Severity.ERROR;
         }
     }
+
+    // duplicate code, trying to call RuntimeInformation produces a pom induced cycle.
+    private String getMavenVersion() {
+        Properties properties = new Properties();
+
+        try (InputStream resourceAsStream =
+                getClass().getResourceAsStream("/org/apache/maven/messages/build.properties")) {
+
+            if (resourceAsStream != null) {
+                properties.load(resourceAsStream);
+            }
+        } catch (Exception e) {
+            System.err.println("Unable determine version from JAR file: " + e.getMessage());
+        }
+
+        return properties.getProperty("version");
```

Comment:
As `DefaultModelValidator` is singleton we can cache this value in filed.

Reply:
ok, method is used only during error message generating, so not necessary to cache